scripts/exit.py

```diff
@@ -4,19 +4,435 @@
+    )
+    for tx_receipt in receipts:
+        path = rlp.encode(tx_receipt["transactionIndex"])
+        receipts_trie[path] = serialize_receipt(tx_receipt)
+
+    key = rlp.encode(burn_tx_receipt["transactionIndex"])
+    proof = receipts_trie.get_proof(key)
+
+    assert (
+        receipts_trie.root_hash == burn_tx_block["receiptsRoot"]
+    ), "Receipts trie root is incorrect"
+
+    return key, proof
+
+
+def find_log_index(burn_tx_receipt: TxReceipt) -> int:
+    """Retrieve the index of the burn event log."""
+    ERC20_TRANSFER_EVENT_SIG = keccak256(b"Transfer(address,address,uint256)")
+    for idx, log in enumerate(burn_tx_receipt["logs"]):
+        topics = log["topics"]
+        if topics[0] == ERC20_TRANSFER_EVENT_SIG and topics[2] == HexBytes(0) * 32:
+            return idx
+
+    # this should not be reached
+    raise Exception("Transfer Log Event Not Found in Burn Tx Receipt")
+
+
+def encode_payload(
+    header_block_number: int,
+    block_proof: List[bytes],
+    block_number: int,
+    timestamp: int,
+    transactions_root: bytes,
+    receipts_root: bytes,
+    burn_tx_receipt: TxReceipt,
+    receipt_proof: List[bytes],
+    path: bytes,
+    log_index: int,
+) -> bytes:
+    """RLP encode the data required to form the calldata for exiting."""
+    payload = [
+        header_block_number,
+        b"".join(block_proof),
+        block_number,
+        timestamp,
+        transactions_root,
+        receipts_root,
+        serialize_receipt(burn_tx_receipt),
+        rlp.encode(receipt_proof),
+        HexBytes(0) + path,
+        log_index,
+    ]
+    return rlp.encode(payload)
+
+
+def build_calldata(burn_tx_id: str = MATIC_BURN_TX_ID) -> bytes:
+    """Generate the calldata required for withdrawing ERC20 asset on Ethereum."""
+    assert is_burn_checkpointed()
+
+    burn_tx, burn_tx_receipt, burn_tx_block = fetch_burn_tx_data(burn_tx_id)
+    start, end, header_block_number = fetch_block_inclusion_data(burn_tx_block["number"])
+    block_proof = build_block_proof(start, end, burn_tx_block["number"])
+    path, receipt_proof = build_receipt_proof(burn_tx_receipt, burn_tx_block)
+
+    log_index = find_log_index(burn_tx_receipt)
+
+    calldata = encode_payload(
+        header_block_number,
+        block_proof,
+        burn_tx_block["number"],
+        burn_tx_block["timestamp"],
+        burn_tx_block["transactionsRoot"],
+        burn_tx_block["receiptsRoot"],
+        burn_tx_receipt,
+        receipt_proof,
+        path,
+        log_index,
+    )
+    return calldata
+
+
+def exit():
+    print("Building Calldata")
+    calldata = build_calldata()
+
+    deployment_addrs = fetch_deployment_data()["Main"]["POSContracts"]
+    root_chain_mgr_proxy_addr, root_chain_mgr_addr = (
+        deployment_addrs["RootChainManagerProxy"],
+        deployment_addrs["RootChainManager"],
+    )
+    root_chain_mgr = Contract.from_explorer(
+        root_chain_mgr_proxy_addr, root_chain_mgr_addr, silent=True
+    )
```

Comment:
```suggestion
    root_chain_mgr = Contract(root_chain_mgr_proxy_addr)
```

Adding `autofetch_sources: True` to the config file will allow us to do away with `from_explorer` and speed things up due to caching.  Also, when testing I ran into an issue in that matic's json providing the addresses points at an out-of-date implementation contract.  So better to just dynamically determine, I think.